frontend/src/views/SQR/Form.vue

```diff
@@ -348,6 +356,27 @@ export default class Form extends Vue {
 
   debounceUpdateForm = _.debounce(this.updateForm, 150);
 
+  async canSubmitForm() {
+    const {
+      params: { fileName, token }
+    } = this.$route;
+    const response = (await firebase
+      .database()
+      .ref(`SQR/allotments`)
+      .orderByChild("token")
+      .equalTo(token)
+      .once("value")).val();
+    const sqrStatus = _.get<string>(response, `${fileName}.status`, "");
+    if (!sqrStatus) {
+      this.allotmentError =
+        "This allotment is not valid, please contact coordinator.";
+    } else if (!this.isCoordinator && sqrStatus.toLowerCase() === "done") {
```

Comment:
I'm not sure it's worth adding `isCoordinator` property that is used only once in the code. Why not to query the user claims here directly? Or even better, add `isCoordinator` to the `user` store module and map it from there.

Reply:
Where would the initial `isCoordinator` be set in the vuex store? Maybe the vuex commit that runs on [auth state change](https://github.com/PureBhaktiArchive/audioseva/blob/master/frontend/src/main.ts#L59) can be replaced with an action that sets the currentUser and does additional stuff like setting isCoordinator?

Reply:
ok, let's leave it as of now, but please consider refactoring this whenever you have time. I have added it as a point to https://trello.com/c/vKAD3ViL/17-load-user-info-from-the-routes.